src/Http/Handlers/AuthenticationHandler.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace HelpScout\Api\Http\Handlers;
+
+use HelpScout\Api\Exception\AuthenticationException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class AuthenticationHandler
+{
+    public function __invoke(callable $handler)
+    {
+        return function (RequestInterface $request, array $options = []) use ($handler) {
+            return $handler($request, $options)->then(
+                function (ResponseInterface $response) use ($request) {
+                    if ($response->getStatusCode() === 401) {
+                        throw new AuthenticationException('Authentication Error', $request, $response);
```

Comment:
What do you about changing the exception message to be `Invalid credentials or expired refresh token`

Reply:
I'm down to change this, although if both are possible returns it feels like the message should be inclusive to both. What about `Not Authorized`, as we use that verbiage in the [docs](https://developer.helpscout.com/mailbox-api/overview/status_codes/)?

Reply:
I like the idea of providing more guidance when/where we can, but am not opposed to staying generic since there are edge cases we may not be aware of.